src/samples/coap/lwm2m-bs-server.c

```diff
@@ -0,0 +1,202 @@
+#define SECURITY_SERVER_SERVER_URI_RES_ID (0)
+#define SECURITY_SERVER_IS_BOOTSTRAP_RES_ID (1)
+#define SECURITY_SERVER_SERVER_ID_RES_ID (10)
+#define SECURITY_SERVER_CLIENT_HOLD_OFF_TIME_RES_ID (11)
+#define SECURITY_SERVER_BOOTSTRAP_SERVER_ACCOUNT_TIMEOUT_RES_ID (12)
+
+const char *known_clients[] = { "cli1", "cli2", NULL };
+
+static struct sol_blob server_one_addr = {
+    .type = &SOL_BLOB_TYPE_NO_FREE,
+    .parent = NULL,
+    .mem = (void *)"coap://localhost:5683",
+    .size = sizeof("coap://localhost:5683") - 1,
+    .refcnt = 1
+};
+
+static struct sol_blob binding = {
+    .type = &SOL_BLOB_TYPE_NO_FREE,
+    .parent = NULL,
+    .mem = (void *)"U",
+    .size = sizeof("U") - 1,
+    .refcnt = 1
+};
+
+static void
+write_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    const char *name = sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo);
+
+    if (response_code != SOL_COAP_RESPONSE_CODE_CHANGED) {
+        fprintf(stderr, "The client %s could not write the object/resource at %s.\n",
+            name, path);
+        return;
+    }
+
+    printf("The client %s wrote the object/resource at %s.\n", name, path);
+
+    sol_lwm2m_bootstrap_server_send_finish(server, bs_cinfo);
+}
+
+static void
+delete_all_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    const char *name = sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo);
+    int r;
+    uint16_t i;
+    struct sol_lwm2m_resource sec_server_one[3];
+    struct sol_lwm2m_resource server_one[3];
+
+    if (response_code != SOL_COAP_RESPONSE_CODE_DELETED) {
+        fprintf(stderr, "The client %s could not delete the object at %s.\n",
+            name, path);
+        return;
+    }
+
+    printf("The client %s deleted the object at %s.\n", name, path);
+
+    // Server One's Security Object
+    SOL_LWM2M_RESOURCE_INIT(r, &sec_server_one[0], SECURITY_SERVER_SERVER_URI_RES_ID, 1,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &server_one_addr);
+    if (r < 0) {
+        fprintf(stderr, "Could not init Security Object's [Server URI] resource\n");
+        return;
+    }
+    SOL_LWM2M_RESOURCE_INIT(r, &sec_server_one[1], SECURITY_SERVER_IS_BOOTSTRAP_RES_ID, 1,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_BOOL, false);
+    if (r < 0) {
+        fprintf(stderr, "Could not init Security Object's [Bootstrap Server] resource\n");
+        return;
+    }
+    SOL_LWM2M_RESOURCE_INT_INIT(r, &sec_server_one[2], SECURITY_SERVER_SERVER_ID_RES_ID, 101);
+    if (r < 0) {
+        fprintf(stderr, "Could not init Security Object's [Short Server ID] resource\n");
+        return;
+    }
+
+    sol_lwm2m_bootstrap_server_write(server, bs_cinfo, "/0/0",
+        sec_server_one, sol_util_array_size(sec_server_one), NULL, NULL);
```

Comment:
It would be nice to register a callback here to check if the operation was successful